performance-tests/bench_2/worker/SetCftParametersAction.cpp

```diff
@@ -0,0 +1,141 @@
+#include "SetCftParametersAction.h"
+
+#include "MemFunHandler.h"
+
+namespace {
+
+uint32_t one_at_a_time_hash(const uint8_t* key, size_t length) {
+  size_t i = 0;
+  uint32_t hash = 0;
+  while (i != length) {
+    hash += key[i++];
+    hash += hash << 10;
+    hash ^= hash >> 6;
+  }
+  hash += hash << 3;
+  hash ^= hash >> 11;
+  hash += hash << 15;
+  return hash;
+}
+
+const ACE_Time_Value ZERO(0, 0);
+
+}
+
+namespace Bench {
+
+  SetCftParametersAction::SetCftParametersAction(ACE_Proactor& proactor)
+ : proactor_(proactor), started_(false), stopped_(false)
+ , set_period_(1, 0), max_count_(0), set_call_count_(0)
+{
+}
+
+bool SetCftParametersAction::init(const ActionConfig& config, ActionReport& report,
+  Builder::ReaderMap& readers, Builder::WriterMap& writers, const Builder::ContentFilteredTopicMap& cft_map)
+{
+
+  std::unique_lock<std::mutex> lock(mutex_);
+  Action::init(config, report, readers, writers, cft_map);
+
+  if (!content_filtered_topic_) {
+    std::stringstream ss;
+    ss << "SetCftParametersAction '" << config.name << "' is missing a valid DDS::ContentFilteredTopic_var content_filtered_topic_" << std::flush;
+    throw std::runtime_error(ss.str());
+  }
+
+  std::string name(config.name.in());
+
+  std::random_device r;
+  std::seed_seq seed{r(), one_at_a_time_hash(reinterpret_cast<const uint8_t*>(name.data()), name.size()), r(), r(), r(), r(), r(), r()};
+  mt_ = std::mt19937_64(seed);
+
+  size_t max_count = 0;
+  auto max_count_prop = get_property(config.params, "max_count", Builder::PVK_ULL);
+  if (max_count_prop) {
+    max_count = static_cast<size_t>(max_count_prop->value.ull_prop());
+  }
+  max_count_ = max_count;
+
+  size_t param_count = 0;
+  auto param_count_prop = get_property(config.params, "param_count", Builder::PVK_ULL);
+  if (param_count_prop) {
+    param_count = static_cast<size_t>(param_count_prop->value.ull_prop());
+  }
+  param_count_ = param_count;
+
+  if (!param_count_) {
+    std::stringstream ss;
+    ss << "SetCftParametersAction '" << config.name << "' is missing a valid param_count parameter" << std::flush;
+    throw std::runtime_error(ss.str());
+  }
+
+  // First check frequency as double (seconds)
+  auto write_frequency_prop = get_property(config.params, "write_frequency", Builder::PVK_DOUBLE);
```

Comment:
These parameter names probably ought to get changed from "write_XXX" to "change_XXX" or something similar.